app/controllers/PaypalController.scala

```diff
@@ -190,17 +192,22 @@ class PaypalController(ws: WSClient, paymentServices: PaymentServices, checkToke
     )(MetadataUpdate.apply)(MetadataUpdate.unapply)
   )
 
-  def updateMetadata(countryGroup: CountryGroup) = NoCacheAction.async(parse.form(metadataUpdateForm)) {
-    implicit request =>
-      val paypalService = paymentServices.paypalServiceFor(request)
-      val marketingOptIn = request.body.marketingOptIn
-      val idUser = IdentityId.fromRequest(request)
-      val contributor = request.session.data.get("email") match {
-        case Some(email) => paypalService.updateMarketingOptIn(email, marketingOptIn, idUser).value
-        case None => Future.successful(Logger.error("email not found in session while trying to update marketing opt in"))
-      }
-      contributor.map { _ =>
-        Redirect(routes.Contributions.thanks(countryGroup).url, SEE_OTHER)
-      }
+  def updateMetadata(countryGroup: CountryGroup) = NoCacheAction.async(parse.form(metadataUpdateForm)) { implicit request =>
+    val paypalService = paymentServices.paypalServiceFor(request)
+    val marketingOptIn = request.body.marketingOptIn
+    val idUser = IdentityId.fromRequest(request)
+    val contributor = request.session.data.get("email") match {
```

Comment:
Another approach without having to extract the `.value` of the update result (which seems like an anti-pattern to me):

```
val email =  request.session.data.get("email")

EitherT.fromOption[Future](email, "email not found in session while trying to update marketing opt in")
  .flatMap(paypalService.updateMarketingOptIn(_, marketingOptIn, idUser))
  .fold(message => Logger.error(message), _ => ())
  .map(_ => Redirect(url, SEE_OTHER))
```